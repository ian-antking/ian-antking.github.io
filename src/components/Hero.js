import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import devices from '../styles/devices'

const HeroImage = styled.div`
  background-image: linear-gradient(
      ${(props) => `${props.theme.shadow}${props.image ? '80' : 'FF'}`},
      ${(props) => `${props.theme.shadow}${props.image ? '80' : 'FF'}`}
    ),
    url(${(props) => props.image});
  height: 40vh;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;

  @media ${devices.mobileL} {
    align-items: flex-end;
    justify-content: center;
  }

  clip-path: polygon(0 0, 100% 0, 100% 100%, 0 ${(props) => props.angle}%);
`;

const Title = styled.h1`
  margin: 0;
  color: ${(props) => props.theme.light};
  font-size: 4em;
`;

const SubTitle = styled.h2`
  margin: 0;
  font-size: 2em;
  color: ${(props) => props.theme.highLight};
`;

const TitleBox = styled.div`
  @media ${devices.mobileL} {
    margin: 10vw;
  }
`;

const Hero = ({ title, subtitle, image, angle }) => {
  const lines = subtitle.split('\\n');
  return (
    <>
      <HeroImage data-testid="hero-image" image={image} angle={angle}>
        <TitleBox>
          <Title>{title}</Title>
          {lines.map((line) => {
            return <SubTitle key={line}>{line}</SubTitle>;
          })}
        </TitleBox>
      </HeroImage>
    </>
  );
};

Hero.defaultProps = {
  image: '',
  angle: 60 + Math.floor(Math.random() * 20),
};

Hero.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  image: PropTypes.string,
  angle: PropTypes.number,
};

export default Hero;
