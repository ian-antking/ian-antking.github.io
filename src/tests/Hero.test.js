import React from 'react';
import { render } from '@testing-library/react';
import Hero from '../components/Hero';

describe('Hero', () => {
  it('renders the correct snapshot', () => {
    const { asFragment } = render(
      <Hero title="test-title" subtitle="test-subtitle" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
