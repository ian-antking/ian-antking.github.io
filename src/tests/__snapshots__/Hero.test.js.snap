// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders the correct snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm fAgQzw"
    data-testid="hero-image"
  >
    <div
      class="sc-AxhCb fsZeIF"
    >
      <h1
        class="sc-AxirZ bjMeBQ"
      >
        test-title
      </h1>
      <h2
        class="sc-AxiKw kojdAC"
      >
        test-subtitle
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
